pkg/authentication/handlers.go

```diff
@@ -44,35 +49,68 @@ func NewAuthenticationConfig(enabled bool, issuer, clientID, clientSecret, redir
 	c.redirectURL = redirectURL
 	c.verificationStrategy = verificationStrategy
 	c.scopes = scopes
-	c.nonceTTL = nonceTTL
+	c.AuthCookieTTLSeconds = cookieTTLSeconds
 
 	return c
 }
 
 type API struct {
-	apiKey           string
-	payloadValidator validation.PayloadValidatorInterface
-	oauth2           OAuth2ContextInterface
-
-	tracer   trace.Tracer
-	logger   logging.LoggerInterface
-	nonceTTL time.Duration
+	apiKey                string
+	payloadValidator      validation.PayloadValidatorInterface
+	oauth2                OAuth2ContextInterface
+	helper                OAuth2HelperInterface
+	cookieManager         AuthCookieManagerInterface
+	authCookiesTTLSeconds int
+
+	tracer trace.Tracer
+	logger logging.LoggerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	mux.Get("/api/v0/login", a.handleLogin)
+	mux.Get("/api/v0/auth/callback", a.handleCallback)
 }
 
 func (a *API) handleLogin(w http.ResponseWriter, r *http.Request) {
 	// add the Otel HTTP Client
 	r = r.WithContext(OtelHTTPClientContext(r.Context()))
 
-	code := r.URL.Query().Get("code")
+	nonce := a.helper.RandomURLString()
+	state := a.helper.RandomURLString()
+
+	ttl := time.Duration(a.authCookiesTTLSeconds) * time.Second
+
+	a.cookieManager.SetNonceCookie(w, nonce, ttl)
+	a.cookieManager.SetStateCookie(w, state, ttl)
+
+	a.oauth2.LoginRedirect(w, r, nonce, state)
```

Comment:
nitpick: I would prefer if this function only returned the URL and the handler wrote to the response the redirect. Mainly for clear separation of responsibility. No problem with keeping it as it is.